ec/ecresource/deploymentresource/elasticsearch/v2/elasticsearch.go

```diff
@@ -0,0 +1,413 @@
+	var es *ElasticsearchTF
+
+	if esObj.IsNull() || esObj.IsUnknown() {
+		return nil, nil
+	}
+
+	if diags := tfsdk.ValueAs(ctx, esObj, &es); diags.HasError() {
+		return nil, diags
+	}
+
+	if es == nil {
+		return nil, nil
+	}
+
+	if es == nil {
+		var diags diag.Diagnostics
+		diags.AddError("Elasticsearch payload error", "cannot find elasticsearch data")
+		return nil, diags
+	}
+
+	templatePayload := utils.EnrichElasticsearchTemplate(utils.EsResource(template), dtID, version, useNodeRoles)
+
+	payload, diags := es.Payload(ctx, templatePayload, skipTopologies)
+	if diags.HasError() {
+		return nil, diags
+	}
+
+	return payload, nil
+}
+
+func ReadElasticsearches(in []*models.ElasticsearchResourceInfo, remotes *models.RemoteResources) (*Elasticsearch, error) {
+	for _, model := range in {
+		if util.IsCurrentEsPlanEmpty(model) || utils.IsEsResourceStopped(model) {
+			continue
+		}
+		es, err := ReadElasticsearch(model, remotes)
+		if err != nil {
+			return nil, err
+		}
+		return es, nil
+	}
+
+	return nil, nil
+}
+
+func ReadElasticsearch(in *models.ElasticsearchResourceInfo, remotes *models.RemoteResources) (*Elasticsearch, error) {
+	var es Elasticsearch
+
+	if util.IsCurrentEsPlanEmpty(in) || utils.IsEsResourceStopped(in) {
+		return &es, nil
+	}
+
+	if in.Info.ClusterID != nil && *in.Info.ClusterID != "" {
+		es.ResourceId = in.Info.ClusterID
+	}
+
+	if in.RefID != nil && *in.RefID != "" {
+		es.RefId = in.RefID
+	}
+
+	if in.Region != nil {
+		es.Region = in.Region
+	}
+
+	plan := in.Info.PlanInfo.Current.Plan
+	var err error
+
+	topologies, err := ReadElasticsearchTopologies(plan)
+	if err != nil {
+		return nil, err
+	}
+	es.setTopology(topologies)
+
+	if plan.AutoscalingEnabled != nil {
+		es.Autoscale = ec.String(strconv.FormatBool(*plan.AutoscalingEnabled))
+	}
+
+	if meta := in.Info.Metadata; meta != nil && meta.CloudID != "" {
+		es.CloudID = &meta.CloudID
+	}
+
+	es.HttpEndpoint, es.HttpsEndpoint = converters.ExtractEndpoints(in.Info.Metadata)
+
+	es.Config, err = ReadElasticsearchConfig(plan.Elasticsearch)
+	if err != nil {
+		return nil, err
+	}
+
+	clusters, err := ReadElasticsearchRemoteClusters(remotes.Resources)
+	if err != nil {
+		return nil, err
+	}
+	es.RemoteCluster = clusters
+
+	extensions, err := ReadElasticsearchExtensions(plan.Elasticsearch)
+	if err != nil {
+		return nil, err
+	}
+	es.Extension = extensions
+
+	accounts, err := ReadElasticsearchTrustAccounts(in.Info.Settings)
+	if err != nil {
+		return nil, err
+	}
+	es.TrustAccount = accounts
+
+	externals, err := ReadElasticsearchTrustExternals(in.Info.Settings)
+	if err != nil {
+		return nil, err
+	}
+	es.TrustExternal = externals
+
+	return &es, nil
+}
+
+func (es *ElasticsearchTF) Payload(ctx context.Context, res *models.ElasticsearchPayload, skipTopologies bool) (*models.ElasticsearchPayload, diag.Diagnostics) {
```

Comment:
Doesn't need to be exported

```suggestion
func (es *ElasticsearchTF) payload(ctx context.Context, res *models.ElasticsearchPayload, skipTopologies bool) (*models.ElasticsearchPayload, diag.Diagnostics) {
```